tsdb/chunks/chunks.go

```diff
@@ -291,24 +314,33 @@ func MergeChunks(a, b chunkenc.Chunk) (*chunkenc.XORChunk, error) {
 			aok = ait.Next()
 			bok = bit.Next()
 		}
-	}
-	for aok {
-		at, av := ait.At()
-		app.Append(at, av)
-		aok = ait.Next()
-	}
-	for bok {
-		bt, bv := bit.At()
-		app.Append(bt, bv)
-		bok = bit.Next()
+		if first {
+			mint = minOf(at, bt)
+			first = false
+		}
+		maxt = minOf(at, bt)
+		numSamples++
+		if numSamples == limit {
```

Comment:
```suggestion
		if numSamples >= limit {
```
Is usually a more readable choice